app/controllers/plans_controller.rb

```diff
@@ -3,7 +3,7 @@ class PlansController < ApplicationController
   require 'pp'
   helper PaginableHelper
   helper SettingsTemplateHelper
-  after_action :verify_authorized
+  after_action :verify_authorized, except: [:overview]
```

Comment:
I think you may have forgotten to remove this exception. You're calling authorize in the method.

Reply:
This expect has been placed intentionally. I don't need pundit to verify that authorize has been called which will fail when trying to find a plan for a non-existing id, raising ActiveRecord::RecordNotFound and therefore never executing authorize which Pundit complains. The three possible test cases I can think of has been added at test/functional/plans_controller_test.rb

Reply:
by default, Rails displays the detailed error message only in development mode in these RecordNotFound scenarios. We have configured roadmap-stg to do the same to make it easier for us to UAT and record bugs. In production though Rails sends the user to a generic 404 page (see screenshot). Since most of our controllers do not have specific handling for RecordNotFound it would be better to do something more generic like update the default 404 page to fit within the styling of the site (we should do this anyway) or add something in the application_controller.

![screen shot 2017-12-05 at 10 48 29 am](https://user-images.githubusercontent.com/1204467/33624643-dfcfc0d2-d9a9-11e7-8912-557d9cdb1103.png)
404 screen when running the app w/ `bundle exec rails s -e production`

Reply:
I think moving the Record Not found exception handling code out of the controller method into the application controller might be better and make it reusable too. For example using Rails "rescue_from" method,
https://stackoverflow.com/questions/13434523/general-rescue-throughout-controller-when-id-not-found-ror

Reply:
@jollopre if you could please remove the RecordNotFound check in the overview method and the pundit exception I'll create a separate ticket for this and add it to the bugfix column. After having seen the above version of what the default Rails 404 page is doing I think we should add it as an MVP task. Would be better to have a themed 404 page than the one above.

Reply:
Rails in production is automatically rescuing from ActiveRecord::RecordNotFound and instead of throwing 5XX status code (which means something not available now, basically), it changes to be 404 (which means I don't have that resource). By removing that line, you have a misleading response since and that's the reason I explicitly recover from that to tell the user that plan didn't exist. I think we should keep this like it is until we don't have a proper handler in place acting at the application_controller.rb level and rescuing from there any Expection which is appropriated message.

Reply:
ok. will accept this as is but please do not introduce more of these RecordNotFound rescues in future PRs. It just adds to the amount of work that will need to be done when addressing #899